File: src/main/java/fi/aalto/cs/apluscourses/intellij/model/ProjectModuleSource.java
package fi.aalto.cs.apluscourses.intellij.model;

import com.intellij.openapi.module.Module;
import com.intellij.openapi.module.ModuleManager;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ProjectModuleSource {

  @NotNull
  public Module[] getModules(@NotNull Project project) {
    return ModuleManager.getInstance(project).getModules();
  }

  @Nullable
  public Module getModule(@NotNull Project project, @NotNull String moduleName) {
    return ModuleManager.getInstance(project).findModuleByName(moduleName);
  }

}

File: src/main/kotlin/fi/aalto/cs/apluscourses/utils/ProjectModuleSource.kt
package fi.aalto.cs.apluscourses.utils

import com.intellij.openapi.module.Module
import com.intellij.openapi.module.ModuleManager
import com.intellij.openapi.project.Project

object ProjectModuleSource {
    fun getModules(project: Project): Array<Module> {
        return ModuleManager.getInstance(project).modules
    }

    fun getModule(project: Project, moduleName: String): Module? {
        return ModuleManager.getInstance(project).findModuleByName(moduleName)
    }
}
